app/models/whitehall_imported_asset.rb

```diff
@@ -3,4 +3,12 @@
 # Represents the raw import of an asset from Whitehall Publisher and
 # the import status of the asset into Content Publisher
 class WhitehallImportedAsset < ApplicationRecord
+  belongs_to :whitehall_import
+
+  belongs_to :image_revision, class_name: "Image::Revision", optional: true
+
+  def variants
+    # Stored as a stringified hash - need to turn it back to JSON
+    JSON.parse(read_attribute("variants").gsub("=>", ":"))
+  end
```

Comment:
We shouldn't need to do any of this ActiveRecord will convert the JSON automatically to a hash. I'm assuming you hit an issue that led to this. 

Just as an example:

```
irb(main):009:0> cr = FactoryBot.create(:content_revision, contents: { "key_1" => "value_1", "key_2" => "value_2" })
   (0.2ms)  BEGIN
  User Create (0.4ms)  INSERT INTO "users" ("name", "email", "uid", "created_at", "updated_at") VALUES ($1, $2, $3, $4, $5) RETURNING "id"  [["name", "John Smith"], ["email", "someone@example.com"], ["uid", "65df8608-5519-4095-bb7f-f9f661e9cb5e"], ["created_at", "2019-12-17 19:39:41.096261"], ["updated_at", "2019-12-17 19:39:41.096261"]]
   (1.1ms)  COMMIT
   (0.1ms)  BEGIN
  ContentRevision Create (0.3ms)  INSERT INTO "content_revisions" ("contents", "created_at", "created_by_id") VALUES ($1, $2, $3) RETURNING "id"  [["contents", "{\"key_1\":\"value_1\",\"key_2\":\"value_2\"}"], ["created_at", "2019-12-17 19:39:41.099551"], ["created_by_id", 51011]]
   (0.2ms)  COMMIT
=> #<ContentRevision id: 15771, title: nil, base_path: nil, summary: nil, contents: {"key_1"=>"value_1", "key_2"=>"value_2"}, created_at: "2019-12-17 19:39:41", created_by_id: 51011>
irb(main):011:0> cr.reload
  ContentRevision Load (0.4ms)  SELECT "content_revisions".* FROM "content_revisions" WHERE "content_revisions"."id" = $1 LIMIT $2  [["id", 15771], ["LIMIT", 1]]
=> #<ContentRevision id: 15771, title: nil, base_path: nil, summary: nil, contents: {"key_1"=>"value_1", "key_2"=>"value_2"}, created_at: "2019-12-17 19:39:41", created_by_id: 51011>
irb(main):012:0> cr.contents
=> {"key_1"=>"value_1", "key_2"=>"value_2"}
irb(main):013:0> cr.contents.class
=> Hash
```

Reply:
You're right, I was facing issues that forced me to write this - something along the lines of "expected these to match, but they didn't":

```diff
- "{ :foo => \"bar\" }"
+ "{ \"foo\": \"bar\" }"
```

But I've gone back and rebased, and removed this method and the tests are passing! 🤷‍♂

Reply:
Curious - it just happened again!

`govuk-docker-run bundle exec rake` passes, but if I run `govuk-docker-run bundle exec rspec spec/lib/whitehall_importer/create_image_revision_spec.rb` in isolation, I get this:

```
  2) WhitehallImporter::CreateImageRevision#call Valid image is provided should create a WhitehallImportedAsset record
     Failure/Error: expect(WhitehallImportedAsset.last.variants).to eq(whitehall_image["variants"])
     
       expected: {"s960"=>"https://assets.publishing.service.gov.uk/government/uploads/s960_valid-image.jpg"}
            got: "{\"s960\"=>\"https://assets.publishing.service.gov.uk/government/uploads/s960_valid-image.jpg\"}"
```

Reply:
For the record, this was down to me using `t.string :variants, json: true, default: "{}"` in the migration. It should have been `t.json :variants, default: {}, null: false`.